sn_client/src/file_apis.rs

```diff
@@ -165,7 +169,9 @@ impl Files {
         let payment = payment_proofs.get(&address.name().0).cloned();
         // TODO: re-enable requirement to always provide payment proof
         //.ok_or(super::Error::MissingPaymentProof(address))?;
-        let closest_peers = self.client.get_closest_local_peers().await?;
+        let all_peers = self.client.get_all_local_peers().await?;
+        let dst = NetworkAddress::from_chunk_address(address);
+        let closest_peers = sort_peers_by_address(all_peers, &dst, CLOSE_GROUP_SIZE)?;
```

Comment:
Should we call `get_closest_local_peers(chunk_address)` here? It should provide the sorted, closest peers to the chunk directly?

Reply:
And hence a smaller `Vec<peers>`?